pkg/operation/botanist/component/hvpa/hvpa.go

```diff
@@ -163,6 +159,43 @@ func (h *hvpa) Deploy(ctx context.Context) error {
 				Namespace: serviceAccount.Namespace,
 			}},
 		}
+		role = &rbacv1.Role{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      roleName,
+				Namespace: h.namespace,
+			},
+			Rules: []rbacv1.PolicyRule{
+				{
+					APIGroups:     []string{"coordination.k8s.io"},
+					Resources:     []string{"leases"},
+					ResourceNames: []string{"hvpa-controller"},
```

Comment:
In my local setup, hvpa-controller fails to create the lease:
```
E0424 07:32:18.321066       1 leaderelection.go:334] error initially creating leader election record: leases.coordination.k8s.io is forbidden: User "system:serviceaccount:garden:hvpa-controller" cannot create resource "leases" in API group "coordination.k8s.io" in the namespace "garden"
E0424 07:32:21.695867       1 leaderelection.go:334] error initially creating leader election record: leases.coordination.k8s.io is forbidden: User "system:serviceaccount:garden:hvpa-controller" cannot create resource "leases" in API group "coordination.k8s.io" in the namespace "garden"
E0424 07:32:24.806254       1 leaderelection.go:334] error initially creating leader election record: leases.coordination.k8s.io is forbidden: User "system:serviceaccount:garden:hvpa-controller" cannot create resource "leases" in API group "coordination.k8s.io" in the namespace "garden"
```

According to https://kubernetes.io/docs/reference/access-authn-authz/rbac/#referring-to-resources, it is not possible to restrict `create` with `resourceNames`. I believe you are also able to reproduce this issue (if you already have existing setup, delete the lease and make sure that it won't be able to create a new one).

Reply:
Suggestion:

```diff
diff --git a/pkg/operation/botanist/component/hvpa/hvpa.go b/pkg/operation/botanist/component/hvpa/hvpa.go
index d52b85e77..2d665c642 100644
--- a/pkg/operation/botanist/component/hvpa/hvpa.go
+++ b/pkg/operation/botanist/component/hvpa/hvpa.go
@@ -165,11 +165,16 @@ func (h *hvpa) Deploy(ctx context.Context) error {
                                Namespace: h.namespace,
                        },
                        Rules: []rbacv1.PolicyRule{
+                               {
+                                       APIGroups: []string{"coordination.k8s.io"},
+                                       Resources: []string{"leases"},
+                                       Verbs:     []string{"create"},
+                               },
                                {
                                        APIGroups:     []string{"coordination.k8s.io"},
                                        Resources:     []string{"leases"},
                                        ResourceNames: []string{"hvpa-controller"},
-                                       Verbs:         []string{"create", "get", "watch", "update"},
+                                       Verbs:         []string{"get", "watch", "update"},
                                },
                                {
                                        APIGroups: []string{""},
```

Reply:
Oh, wow, great find! This was how things were in the beginning, I just didn't understand why and changed it to a tighter scope limited by name. Now I learned why it was like this in the first place. Thanks!

Reply:
Deleted the existing lease on my Seed and saw the above issue, changed the permissions and validated that it now works correctly. Thanks!